doc/source/whatsnew/v0.21.0.txt

```diff
@@ -369,6 +369,7 @@ Reshaping
 - Fixes regression from 0.20, :func:`Series.aggregate` and :func:`DataFrame.aggregate` allow dictionaries as return values again (:issue:`16741`)
 - Fixes dtype of result with integer dtype input, from :func:`pivot_table` when called with ``margins=True`` (:issue:`17013`)
 - Bug in :func:`crosstab` where passing two ``Series`` with the same name raised a ``KeyError`` (:issue:`13279`)
+- :func:`Series.argmin`, :func:`Series.argmax`, and their counterparts on ``DataFrame`` and groupby objects work correctly with floating point data that contains infinite values (:issue:`13595`).
```

Comment:
Okay; I wrote something for the breaking API changes section. Ultimately, though, the difference is just that `argmax` used to throw `ValueError` and now throws `TypeError` with string data; is that an appropriate change for the breaking section?

I added a couple of other tests for Series with object dtype. I haven't seen any existing tests break as a result of this change.